enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,624 @@
+  where the tags are defined.
+- OpenShift installer validates the tags defined in `platform.gcp.userTags` and
+  adds these tags to all resources created during installation and also updates
+  `.status.platformStatus.gcp.resourceTags` of the `infrastructure.config.openshift.io`
+- In-cluster operators refers `.status.platformStatus.gcp.resourceTags` of the
+  `infrastructure.config.openshift.io` to add tags to the resources created later.
+
+#### Variation [optional]
+
+### API Extensions
+Enhancement requires below modifications to the mentioned CRDs
+- Add `OrganizationID`, `userLabels` and `userTags` fields to `platform.gcp` of the
+  `installconfigs.install.openshift.io`
+```\h*go\K
+type Platform struct {
+	// organizationID is the organization identifier which
+	// will be used for deriving the tag key's namespaced name.
+	OrganizationID string `json:"organizationID,omitempty"`
+
+	// userLabels has additional keys and values that the installer will add as
+	// labels to all resources that it creates on GCP. Resources created by the
+	// cluster itself may not include these labels. This is a TechPreview feature
+	// and requires setting featureSet to TechPreviewNoUpgrade to configure labels.
+	UserLabels map[string]string `json:"userLabels,omitempty"`
+
+	// userTags has additional keys and values that the installer will add as
+	// tags to all resources that it creates on GCP. Resources created by the
+	// cluster itself may not include these tags. Tag key and tag value should
+	// be the shortnames of the tag key and tag value resource. This is a
+	// TechPreview feature and requires setting featureSet to TechPreviewNoUpgrade
+	// to configure tags.
+	UserTags map[string]string `json:"userTags,omitempty"`
+}
+```
+
+- Add `OrganizationID`, `resourceLabels` and `resourceTags` fields to `status.platformStatus.gcp`
+  of the `infrastructure.config.openshift.io`
+```\h*go\K
+// GCPPlatformStatus holds the current status of the Google Cloud Platform infrastructure provider.
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.resourceLabels) && !has(self.resourceLabels) || has(oldSelf.resourceLabels) && has(self.resourceLabels)",message="resourceLabels may only be configured during installation"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.resourceTags) && !has(self.resourceTags) || has(oldSelf.resourceTags) && has(self.resourceTags)",message="resourceTags may only be configured during installation"
+type GCPPlatformStatus struct {
+	// organizationID is the Organization ID the new GCP resources created for the cluster.
+	// belongs to. To find the Organization ID follow
+	// https://cloud.google.com/resource-manager/docs/creating-managing-organization#retrieving_your_organization_id
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=32
+	// +kubebuilder:validation:Pattern=`^[1-9][0-9]+$`
+	// +optional
+	OrganizationID string `json:"organizationID"`
```

Comment:
So this probably needs to be moved inside `GCPResourceTag` and perhaps changed from `OrganizationID` to `Parent` or `ParentID` so that we can attach tags from projects as well as organizations

Reply:
suggestion incorporated.